lib/ansible/modules/cloud/azure/azure_rm_acr.py

```diff
@@ -0,0 +1,336 @@
+    :return: dict with the state on Azure
+    '''
+    results = dict(
+        id=registry.id,
+        name=registry.name,
+        location=registry.location,
+        admin_user_enabled=registry.admin_user_enabled,
+        sku=registry.sku.name,
+        provisioning_state=registry.provisioning_state,
+        tags=registry.tags
+    )
+    return results
+
+class Actions:
+    NoAction, Create, Update = range(3)
+
+class AzureRMContainerRegistry(AzureRMModuleBase):
+    """Configuration class for an Azure RM container registry resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            admin_user_enabled=dict(
+                type='bool',
+                required=False,
+                default=False
+            ),
+            sku=dict(
+                type='str',
+                required=False,
+                choices=['Classic', 'Basic', 'Standard', 'Premium']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.state = None
+        self.sku = None
+        self.tags = None
+        self._containerregistry_mgmt_client = None
+
+        self.results = dict(changed=False, state=dict())
+
+        super(AzureRMContainerRegistry, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                      supports_check_mode=True,
+                                                      supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+            
+        resource_group = None
+        response = None
+        to_do = Actions.NoAction
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        # Check if the ACR instance already present in the RG
+        if self.state == 'present':
+            response = self.get_acr()
+            self.results['state'] = response
+
+            if not response:
+                to_do = Actions.Create
+            else:
+                self.log('Results : {0}'.format(response))
+                if response['provisioning_state'] == "Succeeded":
+                    to_do = Actions.Update
+                else:
+                    to_do = Actions.NoAction
```

Comment:
why if provisioning_state is not "Succeeded" you don't update?
will actually create_update_acr() be able to perform update?

Reply:
if the ACR instance was n't successfully created, no update